messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/deployment/JMSConnectionFactoryDefinitionInjectionSource.java

```diff
@@ -420,6 +423,21 @@ static String getDefaulResourceAdapter(DeploymentUnit deploymentUnit) {
                 if (pcf != null && pcf instanceof ConnectionFactoryReferenceFactoryService) {
                     return ((ConnectionFactoryReferenceFactoryService) pcf).getName();
                 }
+                //In case of multiple JNDI entries only the 1st is properly bound
+                if (pcf != null && pcf instanceof ContextListAndJndiViewManagedReferenceFactory) {
+                    ManagedReference ref = ((ContextListAndJndiViewManagedReferenceFactory) pcf).getReference();
+                    Object ra = ref.getInstance();
+                    if (ra instanceof ActiveMQRAConnectionFactoryImpl) {
+                        bindInfo = ContextNames.bindInfoFor(((ActiveMQRAConnectionFactoryImpl) ra).getReference().getClassName());
+                        binder = deploymentUnit.getServiceRegistry().getService(bindInfo.getBinderServiceName());
+                        if (binder != null) {
+                            pcf = binder.getService().getValue();
+                            if (pcf != null && pcf instanceof ConnectionFactoryReferenceFactoryService) {
+                                return ((ConnectionFactoryReferenceFactoryService) pcf).getName();
+                            }
```

Comment:
Minor point; ignore if you want -- this L435-437 block duplicates L423-425 and could be dropped if this entire new code block from L426 was moved above L423.